src/hooks/useValidateTransactionMemo.ts

```
@@ -0,0 +1,136 @@
+
+export const useValidateTransactionMemo = () => {
+  const { network } = useAuthenticationStore();
+  const { transactionXDR } = useTransactionBuilderStore();
+  const { isMemoValidationEnabled } = usePreferencesStore();
+  const { transactionMemo } = useTransactionSettingsStore();
+  const [isValidatingMemo, setIsValidatingMemo] = useState(false);
+  const networkDetails = useMemo(
+    () => mapNetworkToNetworkDetails(network),
+    [network],
+  );
+
+  const shouldValidateMemo = useMemo(
+    () => isMemoValidationEnabled && isMainnet(network),
+    [isMemoValidationEnabled, network],
+  );
+  const [isMemoMissing, setIsMemoMissing] = useState(shouldValidateMemo);
+
+  const checkMemoRequiredFromCache = useCallback(
+    async (txXDR: string, networkType: string): Promise<boolean> => {
+      const response = await cachedFetch<MemoRequiredAccountsApiResponse>(
+        getApiStellarExpertIsMemoRequiredListUrl(),
+        STORAGE_KEYS.MEMO_REQUIRED_ACCOUNTS,
+      );
+
+      // eslint-disable-next-line no-underscore-dangle
+      const memoRequiredAccounts = response._embedded.records || [];
+      const transaction = TransactionBuilder.fromXDR(txXDR, networkType);
+
+      const destination = transaction.operations.find(
+        (operation) => "destination" in operation,
+      )?.destination;
+
+      const matchingBlockedTags = memoRequiredAccounts
+        .filter(({ address }) => address === destination)
+        .flatMap(({ tags }) => tags);
+
+      return matchingBlockedTags.some(
+        (tag) => tag === (TRANSACTION_WARNING.memoRequired as string),
+      );
+    },
+    [],
+  );
+
+  const checkMemoRequiredFromStellarSDK = useCallback(
+    async (txXDR: string, networkUrl: string): Promise<boolean> => {
+      const server = stellarSdkServer(networkUrl);
+      const transaction = TransactionBuilder.fromXDR(txXDR, network);
+
+      try {
+        await server.checkMemoRequired(transaction);
+        return false;
+      } catch (error) {
```

should we check for a specific kind of error here? I see the extension is checking `if ("accountId" in error)` and it looks like we have this _AccountRequiresMemoError_ from the stellar-sdk

then, if it's not the kind of error we are expecting I think we should probably return `false` and print the error with `logger.error()`

what do you think?

---

if you are handling the error inside `checkIsMemoRequired` then you should `throw` here instead of returning `true` to forward the error

---

Had added a comment before, but was looking at the wrong place.  Just disconsider, if you got an email relating to this thread